xml/tuning_dynamic_debug.xml

```diff
@@ -0,0 +1,209 @@
+        <term>Performance tuning</term>
+        <listitem>
+          <para>
+            Use dynamic debugging to monitor and optimize kernel performance by
+            selectively enabling or disabling debugging messages based on the
+            current analysis requirements.
+          </para>
+        </listitem>
+      </varlistentry>
+    </variablelist>
+  </sect1>
+  <sect1 xml:id="sec-tuning-dynamic-debug-check">
+    <title>Checking the status of dynamic debug</title>
+
+    <para>
+      For supported kernel versions that are installed by default, dynamic
+      debug is already enabled. To check the status of dynamic debug, run the
+      following command as the root user:
+    </para>
+
+<screen>&prompt.root;<command>zcat /proc/config.gz | grep CONFIG_DYNAMIC_DEBUG</command></screen>
+
+    <para>
+      If dynamic debug is enabled, you should see an output similar to the
+      following:
+    </para>
+
+<screen>
+CONFIG_DYNAMIC_DEBUG=y
+CONFIG_DYNAMIC_DEBUG_CORE=y
+</screen>
+  </sect1>
+  <sect1 xml:id="sec-tuning-dynamic-debug-use">
+    <title>Using dynamic debug</title>
+
+    <para>
+      To enable specific debug messages or logs within the running kernel, you
+      can use the <command>echo</command> command and write to the
+      <filename>/sys/kernel/debug/dynamic_debug/control</filename> file.
+    </para>
+
+    <para>
+      The following examples illustrate certain simple uses of dynamic debug:
+    </para>
+
+    <note>
+      <para>
+        Dynamic debug relies on specific debugging macros, such as
+        <systemitem>pr_debug</systemitem>, embedded in the kernel code. These
+        macros are used by kernel developers to insert debugging messages into
+        the code.
+      </para>
+      <para>
+        The examples in this section assume that the
+        <systemitem>pr_debug</systemitem> macro works correctly because dynamic
+        debug is allowed for the running kernel.
+      </para>
+    </note>
+
+    <variablelist>
+      <varlistentry>
+        <term>Enabling debug messages for a specific file</term>
+        <listitem>
+          <para>
+            To enable the debug messages for a specific file, use the following
+            example:
+          </para>
+<screen>&prompt.root;<command>echo "file <replaceable>FILE_NAME.c</replaceable> +p" > /sys/kernel/debug/dynamic_debug/control</command></screen>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term>Enabling debug messages for a specific module</term>
+        <listitem>
+          <para>
+            To enable debug messages for a specific kernel module, use the
+            following example:
+          </para>
+<screen>&prompt.root;<command>echo "module <replaceable>MODULE_NAME</replaceable> +p" > /sys/kernel/debug/dynamic_debug/control</command></screen>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term>Enabling debug messages with a specific log level</term>
```

Comment:
This one is interesting! I didn't even knew it existed :-) Can you point me at where you've seen this documented?